api.py

```diff
@@ -0,0 +1,121 @@
+import sys
+import os
+import shutil
+import uuid
+
+from fastapi import FastAPI, File, UploadFile, Form, BackgroundTasks, HTTPException
+from fastapi.responses import FileResponse
+import uvicorn
+from typing import Annotated
+
+import roop.globals
+from roop.core import decode_execution_providers, suggest_execution_threads, limit_resources, start, \
+    get_frame_processors_modules
+from roop.utilities import resolve_relative_path
+
+
+TEMP_FILES_PATH: str
+app = FastAPI()
+
+
+def update_status(message: str) -> None:
+    scope: str = 'ROOP.API'
+    print(f'[{scope}] {message}')
+
+
+def set_init_global_params() -> None:
+    global TEMP_FILES_PATH
+    TEMP_FILES_PATH = resolve_relative_path('../tmp/')
+    if os.path.exists(TEMP_FILES_PATH):
+        shutil.rmtree(TEMP_FILES_PATH)
+        update_status("Initial start. Old temp folder deleted")
+    os.makedirs(TEMP_FILES_PATH)
+
+    roop.globals.headless = True
+
+    execution_provider = os.getenv('EXECUTION_PROVIDER')
+    update_status("execution provider is set to {}".format(execution_provider))
+    if execution_provider == 'CUDA':
+        execution_provider_list = ['cuda']
+    elif execution_provider == 'CPU':
+        execution_provider_list = ['cpu']
+    else:
+        execution_provider_list = ['cpu']
+    roop.globals.execution_providers = decode_execution_providers(execution_provider_list)
+
+    roop.globals.temp_frame_format = os.getenv('TEMP_FRAME_FORMAT')
+    roop.globals.temp_frame_quality = int(os.getenv('TEMP_FRAME_QUALITY'))
+    roop.globals.output_video_quality = int(os.getenv('OUTPUT_VIDEO_QUALITY'))
+    roop.globals.max_memory = int(os.getenv('MAX_MEMORY'))
+
+    update_status("Initial roop parameters set")
+
+
+def remove_request_dir(path:str) -> None:
+    if os.path.exists(path):
+        shutil.rmtree(path)
+        update_status("Temporary request directory {} deleted".format(path))
+
+
+@app.post("/start")
+async def image_file(
+        background_tasks: BackgroundTasks,
```

Comment:
the core.py and api.py should use the same defaults - either we hardcode it in the globals or introduce a get_defaults() method there